t/op/magic.t

```diff
@@ -433,6 +433,26 @@ EOP
   }
 }
 
+# Check that assigning to $0 properly handles upgraded strings:
+{
+  my $eacute = "\N{U+00e9}";
+  utf8::encode($eacute);
+  utf8::upgrade($eacute);
+
+  my $eacute_downgrade = $eacute;
+  utf8::downgrade($eacute_downgrade);
+
+  $0 = $eacute;
+  is ($0, $eacute, 'checked $0');
+
+  SKIP: {
+    skip "Test is for Linux, not $^O" if $^O ne 'linux';
+
+    my $slurp = `cat /proc/$$/cmdline`;
```

Comment:
Do we have any rules about running commands without the full path name?  Do we know that $PATH is safe and this won't run a malicious 'cat'?

Reply:
It could be replaced by an open and readline to avoid the potential issue.

Reply:
done